pywbemtools/pywbemcli/_cmd_connection.py

```diff
@@ -315,13 +336,13 @@ def connection_list(context):
     """
     List the WBEM connection definitions.
 
-    List the WBEM connection definitions in the connections file as a table.
-
-    In the table, the current connection is marked with an "*" after its name.
+    This command displays all entries in the connections file and the
+    current connection if it is not in the connections file as
+    a table using the command line output_format to define the
+    table format.
 
-    Examples:
-
-      pywbemcli connection list
+    An "!" before the name indicates the selected connection.
+    A '!' before the name indicates that it is the current connection.
```

Comment:
Multiple issues:
1. It is not clear what a "selected" connection is other than the new current connection. So I think there is no need to indicate a selected connection separately from the current connection.
2. In the second sentence (about current connection), "!" should be changed to "*".
3. You also introduced the concept of a default connection, in the code it is prefixed with "#" but it is missing in this description.

Reply:
That should be the default connection.  It is different than the current connection.  It is the one we used connection select to define as the default.

That should be 

    An "!" before the name indicates the default connection. See connection select.
    A '*' before the name indicates that it is the current connection.

Reply:
**Discussion point**:

The introduction of a default marker that represents the default connection has all kinds of ugly consequences: Hard to distinguish from current connection, inconsistent behavior (we can now set it but never get rid of it).

How about representing the default connection definition simply as the connection definition named "default"? That allows doing everything with it.

Reply:
Also, the fact that "save" sets the current connection is weird. There is also an issue with it. The following causes two current connections to be displayed:
```
$ pywbemcli                           
Enter 'help' for help, <CTRL-D> or ':q' to exit pywbemcli.
pywbemcli> connection add -m tests/unit/simple_assoc_mock_model.mof mock1
pywbemcli> connection add -m tests/unit/simple_mock_model.mof mock2
pywbemcli> connection list
WBEM server connections: (#: default, *: current)
+--------+-------------+-----------+-------------+----------------------------------------+
| name   | namespace   |   timeout | no-verify   | mock-server                            |
|--------+-------------+-----------+-------------+----------------------------------------|
| mock1  | root/cimv2  |        30 | False       | tests/unit/simple_assoc_mock_model.mof |
| mock2  | root/cimv2  |        30 | False       | tests/unit/simple_mock_model.mof       |
+--------+-------------+-----------+-------------+----------------------------------------+
pywbemcli> connection select mock1
"mock1" current
pywbemcli> connection list
WBEM server connections: (#: default, *: current)
+--------+-------------+-----------+-------------+----------------------------------------+
| name   | namespace   |   timeout | no-verify   | mock-server                            |
|--------+-------------+-----------+-------------+----------------------------------------|
| *mock1 | root/cimv2  |        30 | False       | tests/unit/simple_assoc_mock_model.mof |
| mock2  | root/cimv2  |        30 | False       | tests/unit/simple_mock_model.mof       |
+--------+-------------+-----------+-------------+----------------------------------------+
pywbemcli> connection save mock3
pywbemcli> connection list
WBEM server connections: (#: default, *: current)
+--------+-------------+-----------+-------------+----------------------------------------+
| name   | namespace   |   timeout | no-verify   | mock-server                            |
|--------+-------------+-----------+-------------+----------------------------------------|
| *mock1 | root/cimv2  |        30 | False       | tests/unit/simple_assoc_mock_model.mof |
| *mock3 | root/cimv2  |        30 | False       | tests/unit/simple_assoc_mock_model.mof |
| mock2  | root/cimv2  |        30 | False       | tests/unit/simple_mock_model.mof       |
+--------+-------------+-----------+-------------+----------------------------------------+
pywbemcli> 
```

Reply:
Conclusion is to have '*' for current connection and '#' for default connection.